File: backend/app/services/language_mapper.py
from app.utils.exceptions import InvalidLanguageException

LANGUAGE_MAP = {
    "python": 71,
    "javascript": 63,
    "sql": 82
}

def get_language_id(language: str) -> int:
    lang_key = language.strip().lower()
    if lang_key not in LANGUAGE_MAP:
        raise InvalidLanguageException(language)
    return LANGUAGE_MAP[lang_key]

File: backend/app/services/language_mapper.js
const { InvalidLanguageException } = require('../utils/exceptions');

const LANGUAGE_MAP = {
  "python": 71,
  "javascript": 63,
  "sql": 82
};

function getLanguageId(language) {
  if (!language) {
    throw new InvalidLanguageException(language);
  }
  const langKey = language.trim().toLowerCase();
  if (!(langKey in LANGUAGE_MAP)) {
    throw new InvalidLanguageException(language);
  }
  return LANGUAGE_MAP[langKey];
}

module.exports = {
  LANGUAGE_MAP,
  getLanguageId
};
